### PR TITLE
[TM ONLY] Reviving spessmen with defibs or surgery now forces them into revival shock)

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Getting revived via defibrillator or surgery will put you into revival shock, making you easier to wound for 10 minutes and giving you a major -15 mood debuff, as well as lowering your sanity to crazy. This is being done to change how much players value their lives after beheading changes, as nowadays chance of you getting RRed is extremely slim and you'll most likely get revived within 10-15 minutes unless medbay is destroyed.

Also cleaned up defib code because holy it was bad (374 changed lines, mood)

## Why It's Good For The Game

Removal of beheadings resulted in people being unafraid of death as there's only a few ways they can be round removed now. This change should help antags facing unending tides of constantly-revived validhunters and make such players reconsider their life choices that led them to hunting changelings as an assistant with a blowtorch.

## Changelog
:cl:
add: Spessmen revived via defibs or surgery will now experience revival shock, severely decreasing their mood and sanity and making them easier to wound for some time.
code: Cleaned up defibrillator code.
/:cl:

### CODE REVIEW

**Comment by zoomachina** [comment]
> Removal of beheadings resulted in people being unafraid of death as there's only a few ways they can be round removed now

why not just add beheadings back then, if you think that's the case?

**Comment by Xackii** [comment]
just don't die lol

**Comment by necromanceranne** [comment]
Hypersensitive is about to be one of the meaner quirks to run with this change if you love fighting a lot.

**Comment by ElGitificador** [comment]
this could even make one think beheading removal was an error 🤔

**Comment by SmArtKar** [comment]
> > Removal of beheadings resulted in people being unafraid of death as there's only a few ways they can be round removed now
> 
> why not just add beheadings back then, if you think that's the case?

This is an alternative to bringing beheadings back as there's no consensus on that topic yet. If beheadings ever get brought back, this can be removed.

**Comment by Bibarra** [comment]
Why not have both? Having lasting consequences for death makes death more meaningful.

**Comment by zoomachina** [comment]
> Hypersensitive is about to be one of the meaner quirks to run with this change if you love fighting a lot.

no no, you don't have to pretend like this pr only affects deaths from antags, this affects the typical spaceman much more than it could ever affect validhunters

**Comment by Xackii** [comment]
Make sure the antags don't get this debuff. Ling in particular.

**Comment by SmArtKar** [comment]
> Make sure the antags don't get this debuff. Ling in particular.

Lings are immune to this, everyone else isn't

**Comment by Deultus** [comment]
It's just annoying nonsense that has nothing to do with beheading. characters die easily. and personally, I don't want to die in most situations, but one way or another it happens, and there's nothing pleasant about suffering for 10 minutes in any situation. Decapitation was the antagonists' tool to get you out of the round completely, now there is an alternative to this in the form of destroying the stomach. so why punish the player in any situation by replacing the situation when specifically the antagonist has completely removed you

**Comment by SmArtKar** [comment]
> It's just annoying nonsense that has nothing to do with beheading. characters die easily. and personally, I don't want to die in most situations, but one way or another it happens, and there's nothing pleasant about suffering for 10 minutes in any situation. Decapitation was the antagonists' tool to get you out of the round completely, now there is an alternative to this in the form of destroying the stomach. so why punish the player in any situation by replacing the situation when specifically the antagonist has completely removed you

its not the antagonist's fault you performed what is commonly known a "skill issue" and died. You should value your life, its not disposable and this is a punishment for treating it as such.

**Comment by Pickle-Coding** [comment]
Good change. Always felt weird when someone immediately gets back to 100% with no need for recovery after getting revived, happily dancing and running away instantly when they get out of stasis.

**Comment by zoomachina** [comment]
> Iunno, dying isn't too common in my experience outside of like, miners and shit. Incidental death is a little rarer (and sometimes a lot more permanent) than death from combat.

do you believe that medbay mainly exists to revive validhunters?

**Comment by necromanceranne** [comment]
Okay, listen....

I know that this is is meant to penalize combatants.

But.

_Could Morbid give you a minor mood buff or ignore the mood penalty from being revived?_

My reasoning being that Morbid _has_ an active fascination with death built into the flavour, as do coroners by extension. It makes some degree of sense for them to maybe not be as mentally affected by revival as others, they love it. Since there is one role who is guaranteed to have the trait (coroner), and another who sometimes has it (scythe chaps), it probably isn't going to tip the balance too much. (and easily wounded is already doing most of the heavy lifting anyway)

**Comment by necromanceranne** [comment]
> maybe we should just remove medbay to rub it in for those validhunters?

Let's not start getting hyperbolic here. There is inherent merit to penalizing death in some fashion, and I think this is probably the most fairest method of doing so. 

It can be counteracted (mood stabilizers and mood boosting interactions like drinking and eating), it can be circumvented (easily wounded is mostly just going to make you a lot more conscious about the wounds you're taking and Frail is an extremely common quirk people take because of how it can be worked around if you're careful), and if you do die from an incidental accident, you probably aren't going to feel the effects as much soon enough. Mood doesn't hit nearly that hard for non-combatants.

**Comment by Generubi** [comment]
I think this is the worst possible thing that could've been added, a straight nerf to medbayers just because...

**Comment by SmArtKar** [comment]
> I think this is the worst possible thing that could've been added, a straight nerf to medbayers just because...

this doesn't affect people actually playing med though

**Comment by Majorkite** [comment]
> > I think this is the worst possible thing that could've been added, a straight nerf to medbayers just because...
> 
> this doesn't affect people actually playing med though

yeah im going to love having even more people clogging up the bay hanging around because they are afraid to get hurt outside, or went back out and got wounded by the same antagonist running amok, even more heavy lacerations and running out of blood, third degree and catastrophic burns to treat. Their fault for existing near the antag right?

**Comment by necromanceranne** [comment]
Does anyone have the stats for how often Frail is taken as a quirk? Because the effect of death from revival shock is already a present mechanic anyone can take.

**Comment by EricZilla** [comment]
I kind of agree on it'd be funny if morbid got a small mood buff instead of debuff from dying, but got like, a 15 minute weakness instead so they don't just try to game it

**Comment by necromanceranne** [comment]
I think, on someones else's suggestion, just a theme change for the mood debuff is also fine. (Or, alternatively, I'll just do a followup with an idea I'm already cooking)

**Comment by DaydreamIQ** [comment]
-15 Is kind of absurdly high when most bad ones drop you by less than half that and most good mood bonuses are only a +3 at best. The wounding thing is w/e though, I'd personally just grab a first aid auto pen and go back into the fight

**Comment by Crimsonerva** [comment]
Do you think the mood and/or wound weakness could be nonlinear/scale over time? For example starting off at -20 for 5 minutes, -15 for 3, -10 for 2, etc. I think this would be a great way to achieve the goal of enforcing recovery but doesn't make it overly drastic. People who are immediately recovering (~5 minutes from defib) are debilitated by the high sanity loss, but this way it doesn't "stretch" and suddenly fade away. It also gives incentive to stay in medbay/recovery for a minute, but allows you to leave a bit earlier if necessary (round-ending threat, fighting while stable?).

**Comment by SimplySmiley123** [comment]
> Why not have both? Having lasting consequences for death makes death more meaningful. This is a good change.

And a lot more bullshit then it already tends to be.
Don't need to be screwed over after getting wordlessly and instantaniously murdered in broad daylight. Or at least not more then you likely already ended up.

**Comment by MrMelbert** [comment]
> -15 Is kind of absurdly high when most bad ones drop you by less than half that and most good mood bonuses are only a +3 at best. The wounding thing is w/e though, I'd personally just grab a first aid auto pen and go back into the fight

I suggested it be a very penalizing debuff so you can't just out-mood it. 
There sole reason for the mood penalty is to give you a movespeed and actionspeed slowdown (from sanity), forcing you to take things a bit slower until it wears off.

**Comment by MrMelbert** [comment]
> Do you think the mood and/or wound weakness could be nonlinear/scale over time? 

I think this would work well though. I'm personally a fan of "big debuff that lessens over time", it'd fit well here

File: code/datums/status_effects/debuffs/revival_shock.dm [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/datum/status_effect/revival_shock
+	id = "revival_shock"
+	alert_type = /atom/movable/screen/alert/status_effect/revival_shock
+	duration = 10 MINUTES
+	status_type = STATUS_EFFECT_REFRESH
```
Comment (MrMelbert) [inline code/datums/status_effects/debuffs/revival_shock.dm:5]:
```suggestion
	status_type = STATUS_EFFECT_REFRESH
	remove_on_fullheal = TRUE
	heal_flag_necessary = HEAL_ADMIN // unless you want it on generic HEAL_STATUS. up to you
	
```

Comment (MrMelbert) [reply code/datums/status_effects/debuffs/revival_shock.dm:5]: Also you can put the moodlet application on the station effect apply / remove so ahealing removes the moodlet as well

**Comment by Loafin34** [comment]
10 minutes is too long

**Comment by Formerxboxgamer** [comment]
actual validhunters know how to powergame mood. This doesnt nerf anything other than newer or non powergamer players who die because they're new/dont play to win. Close this and bring back decapping, because as it stands this is just going to be a "You dont get to play the game for 10 minutes" PR for engineers who die to rads while they're learning how to setup the SM.

**Comment by itseasytosee** [comment]
I love the direction this is pointing

**Comment by SomeoneYouProbablyKnow** [comment]
I think we SHOULD have more permanent consequences for combat, not permanent consequences for death. That's what i wished wounds would be but alas. 
But consider that Validhunters are the ones who'll survive fights at 3 HP or drink superhealing chemistry drinks. Their whole shtick is trying their best not to die already. And the things they die to usually don't let them recover (or if they do i'd be worth it in their eyes)

But i don't really see an issue with this except it being rooted in mood. Hate mood.
-15 is very high considering you'll have other moodlets probably from the thing that killed you. It's pretty fair for someone who died and got revived but if you've been dead in medbay for a while and then this happens that's just kicking people who are really down. But that's life

Funnily enough, this makes empath a decent quirk. Just figure out who got tanked mood, good odds they can be wounded.

I'd offer adding something like painkillers preventing this effect (so i can roleplay House). But that's a lot of work and you'd have to make sure you can't do combat while hopped on painkillers through funky effects like shaking and hallucination slowness etc. That would at least give people a choice. Getting saddled with a bad choice feels much better than hjust getting a bad effect.

**Comment by CRITAWAKETS** [comment]
Not gonna lie, the effects of this really won't deter validhunters but would absolutely destroy security, given how their entire combat strategy is built around movespeed. Higher move speed means better regrouping, evasion and the ability to baton joust way, way better.

I like the mood debuff from a flavor standpoint but from a balance standpoint im not really convinced it'll be good for the game.

**Comment by KingkumaArt** [comment]
> actual validhunters know how to powergame mood. This doesnt nerf anything other than newer or non powergamer players who die because they're new/dont play to win. 

This pretty much explains my thoughts on it. 90% of my deaths are from a variant of the area I need to work in being hazardous - e.g. being in a spaced area and being unable to find a crowbar.

Also on the decap being the only real way of RRing people - there are plenty of ways to RR people in practice, while not truly round-removing them. This can include:

- Throwing their stripped corpse into a locker in maints and putting in some room that nobody visits, and then cutting said room's airlock wires
- Shooting it into space with the mass driver along with all the other trash
- Taking it to public mining and throwing it into the middle of the lava river/into the plasma river on icebox
- Just getting a spacesuit and leaving them in the middle of space on a different zlevel

And remember, the gibber board in the service lathe is free, and it works on bodies just as well as it does on a meatball.

**Comment by castawaynont** [comment]
the last thing medbay needs is more dead time and inconvenience imo

**Comment by SimplySmiley123** [comment]
> Not gonna lie, the effects of this really won't deter validhunters but would absolutely destroy security, given how their entire combat strategy is built around movespeed. Higher move speed means better regrouping, evasion and the ability to baton joust way, way better.
> 
> I like the mood debuff from a flavor standpoint but from a balance standpoint im not really convinced it'll be good for the game.

Yeahhh... Validhunters and players completely detached from the character still won't care about this. Either because they're already gaming hard enough to ignore the consequences of their actions or getting booted out of the game not mattering to them whatsoever, or being the goal in the first place.

This would just make those unfair and/or environmental, random deaths even worse to deal with.
Alongside making already loitered by fully healthy players medbay, now loitered with players forced to sit there for a while after getting revived.

**Comment by SomeoneYouProbablyKnow** [comment]
> Not gonna lie, the effects of this really won't deter validhunters but would absolutely destroy security, given how their entire combat strategy is built around movespeed. Higher move speed means better regrouping, evasion and the ability to baton joust way, way better.

Well hey now. Everyone benefits from movespeed, it's everyone's combat strategy. As for sec half the time they don't regroup and instead attack enemies one at a time like in a kung fu movie. Frankly lower movespeed might make them more organized. 

Their main strategy is dying to antags and then being revived. A seccie can die a dozen times an antag can only die once. And yes this puts a big dent in sec's endles meatwave tactics.
Which is good, right now they can solve most problems with brute force and that's boring. Make them actualy do smart things. Cooperate, deputize etc. Why would they do that whrn they can just baton anything they don't like and get endless revives.

tlfr : this will fuck over everyone (which is the point) including sec. Which is good

**Comment by JohnFulpWillard** [comment]
This seems like a potential avenue for future psychologist content and I hope we capitalize on it

**Comment by Majorkite** [comment]
Alright time to better address this now that I'm not running out the door to work and have had a day to think about this.

The wounds: Wounds are lacerations, pierces, fractures, and burns. Half the wounds result in massive blood loss which takes time to replace and significant resources. Burns above second degree will straight up put you in a death loop if untreated with the tube or an augment/amputation. Broken bones are mobility loss in most places and are the least severe to deal with. 

These things already massively complicate an average 'my skill is throw on table and tend wounds' doctor's day, and still bog me down with gathering materials and getting people to stand still. Increasing the frequency not only punishes people who, as above mentioned, simply died to radiation or decompression that can happen at random, but also med. Not worth adding for the amount of needless annoyance it will cause just to fuck up a couple validhunter's days. 

The mood: not as bad, because you can at least get neutral pretty easy if you have a decent quirk setup or know what you're doing, but again punishes the fuck out of people that don't know the ins and outs of the mood system for the crime of dying to literally anything. 

Round removal: The endless discourse over beheading is honestly the most perplexing thing I've seen. Yeah it was an easy way to RR someone, there are also plenty of other ways to RR someone, lots of hard removals above but a multitude of unused extenders of someones time spent in the bay being dead. Burn your bodies. Actually burn them, de-husking someone takes an average of 10 minutes to take care of especially if its early in the round. De-limb, gut them, smash their head until they get a cranial fissure and reach in and yank out their eyes with your bare fucking hands and EAT them, the list goes on. 

Finally, you can discount all of the above and still be wrong about how this is going to affect the balance of the game because tot/wizard/heretic/cult vs validhunters and sec doesnt exist in a vacuum. The other plethora of ways to die to random space bears, spiders, radiation, rods, fires, decompression, whatever is going to disproportionately punish compared to the actual target of this pr. 

I say if you do want to have this in, have it be something exclusive to assistants. Or flesh it out more with counters and more interactions to bypass it if you are sufficiently prepared. 

In short, phrases like

>this will fuck over everyone (which is the point) including sec. Which is good

are things spoken by the utterly deranged. Fucking over everyone for the benefit of antags getting a little more breathing room in valid hunt situations is not good. Its that other thing. Bad.

**Comment by MrMelbert** [comment]
> as it stands this is just going to be a "You dont get to play the game for 10 minutes" PR for engineers who die to rads while they're learning how to setup the SM.

I think you vastly overestimate how much this would make you "unable to play the game". 
Both of the debuffs it gives you are literally just negative quirks that people use regularly.

The average engineer, clown, botanist, bartender, scientist who dies to radiation will be largely unaffected by suddenly acquiring the frail quirk. Everything that slaps an engineer does enough damage to wound anyways, most jobs don't regularly get wounded in their work place, and simple mobs largely don't even apply wounds, so it's not jobs like Shaft Miner would be affected [by the wound debuff in particular] much either. 

The hefty mood debuff can certainly be impactful for jobs like Shaft Miner which rely on mobility, but it's really not the end of the world. Especially since you can get moodlets to offset that (or medbay can provide moodlets such as Musician, which like, sounds like a fun layer on top of medbay gameplay to me? caring for your patients after fixing them up?)

**Comment by MrMelbert** [comment]
Also how are y'all getting +15 from mood events. It can't be that easy can it?

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -887,6 +887,15 @@
 	// The signal is called after everything else so components can properly check the updated values
 	SEND_SIGNAL(src, COMSIG_LIVING_REVIVE, full_heal_flags)
 
+// If we have mood, applies a long-lasting mood debuff, majorly decreases our sanity and making us easier to injure. Does not affect changelings (duh)
+/mob/living/proc/revival_shock()
+	if (isnull(mob_mood) || IS_CHANGELING(src))
```
Comment (MrMelbert) [inline code/modules/mob/living/living.dm:892]: Someone did point out that this gives empaths the ability to ling-check. Jsyk

**Comment by SomeoneYouProbablyKnow** [comment]
> In short, phrases like

> > this will fuck over everyone (which is the point) including sec. Which is good

> are things spoken by the utterly deranged. Fucking over everyone for the benefit of antags getting a little more breathing room in valid hunt situations is not good. Its that other thing. Bad.

ah the jounralist are quoting the tldr again.
Let me spell it out then (or yknow kinda repeat the body of the post instead of a flashy quote)

The PR is trying to add consequences to dying. Validhunters will skirt around that same way they usually do. Most people will be mildly annoyed after being revived. Security on the other hand will be effected disproportionately. THAT specific thing is good because my hope is it would force them to make smart interesting moves instead of being able to baton all their problems away. 

Really, If you're so worried about "the other thing" how about we give everyone cod style health regeneration or bring back cloning, god forbid anyone be inconvenienced. That way no one with a brain will need to work medbay. Win-Win

**Comment by excalisword** [comment]
I think something you could add to this is to make it so that being in a non-bloody room makes the timer on this progress faster, and then you could make it so that being in the same room as the psychologist makes the timer progress twice as fast. Then you could do an inverse for morbid where being in a bloody room makes the timer progress faster and being around the coroner makes it progress twice as fast.

**Comment by SimplySmiley123** [comment]
> > Not gonna lie, the effects of this really won't deter validhunters but would absolutely destroy security, given how their entire combat strategy is built around movespeed. Higher move speed means better regrouping, evasion and the ability to baton joust way, way better.
> 
> Well hey now. Everyone benefits from movespeed, it's everyone's combat strategy. As for sec half the time they don't regroup and instead attack enemies one at a time like in a kung fu movie. Frankly lower movespeed might make them more organized.
> 
> Their main strategy is dying to antags and then being revived. A seccie can die a dozen times an antag can only die once. And yes this puts a big dent in sec's endles meatwave tactics. Which is good, right now they can solve most problems with brute force and that's boring. Make them actualy do smart things. Cooperate, deputize etc. Why would they do that whrn they can just baton anything they don't like and get endless revives.
> 
> tlfr : this will fuck over everyone (which is the point) including sec. Which is good

It fucks over everyone but antagonists and validhunters even more then they tend to get already.

**Comment by Majorkite** [comment]
> > In short, phrases like
> 
> > > this will fuck over everyone (which is the point) including sec. Which is good
> 
> > are things spoken by the utterly deranged. Fucking over everyone for the benefit of antags getting a little more breathing room in valid hunt situations is not good. Its that other thing. Bad.
> 
> ah the jounralist are quoting the tldr again. Let me spell it out then (or yknow kinda repeat the body of the post instead of a flashy quote)
> 
> The PR is trying to add consequences to dying. Validhunters will skirt around that same way they usually do. Most people will be mildly annoyed after being revived. Security on the other hand will be effected disproportionately. THAT specific thing is good because my hope is it would force them to make smart interesting moves instead of being able to baton all their problems away.
> 

That'd be pretty convincing if validhunters weren't explicitly stated to be the target of this whole thing and not the entire security department in general. I put the quote in because it neatly sums up why this entire exercise is bad due to it literally screwing everyone over but the people that it was originally stated to be intended for. I think its not had the effort put into it to make it something really interesting to engage with.

**Comment by KingkumaArt** [comment]
> Also how are y'all getting +15 from mood events. It can't be that easy can it?

Me drinking a beer and eating a meal the chef made by hand

**Comment by Generubi** [comment]
> > I think this is the worst possible thing that could've been added, a straight nerf to medbayers just because...
> 
> this doesn't affect people actually playing med though

The influx of bodies being a lot higher, also having to treat infinite deep wounds/3rd degree burns and things...

**Comment by SmArtKar** [comment]
> Does anyone have the stats for how often Frail is taken as a quirk? Because the effect of death from revival shock is already a present mechanic anyone can take.

image

about in the middle of the set among all quirks

**Comment by projectkepler-ru** [comment]
You're pretty good!!

**Comment by necromanceranne** [comment]
> about in the middle of the set among all quirks

That's pretty reasonably common. enough to suggest that if this was already a problem we would have seen the impact of the quirk already.

**Comment by EricZilla** [comment]
I cant believe we are getting psychiatrist content in 2024, finally, a reason for them to exist

**Comment by Admiral-Gigaton** [comment]
Lot of angry people here who dislike this. I just think the 10 minute effect might be a bit long, 10 minutes is a long time in the grand scheme of the round (unless youre trying to build autism)

**Comment by Majorkite** [comment]
> > about in the middle of the set among all quirks
> 
> That's pretty reasonably common. enough to suggest that if this was already a problem we would have seen the impact of the trait already.

This is enough data to make me think it might not be so bad. I still think the time and severity should be tweaked though. Like others have said 10 minutes is a whole shuttle call. Not an insignificant amount of time.

**Comment by SomeoneYouProbablyKnow** [comment]
> > > about in the middle of the set among all quirks
> > 
> > 
> > That's pretty reasonably common. enough to suggest that if this was already a problem we would have seen the impact of the trait already.
> 
> This is enough data to make me think it might not be so bad. I still think the time and severity should be tweaked though. Like others have said 10 minutes is a whole shuttle call. Not an insignificant amount of time.

That is true but out of the 10 minutes you'll spend like 5 in medbay getting healed after revival, probably more if smart docs prioritize rezzing people to start the 10 min timer. Unless a doc stabs you during surgery or the fighting is in medbay you'll be fine with the extra wound chance as long as you play it semi safe. 10 min of -15 is a lot tho but i'm never sure about mood from just the numbers.

**Comment by norsvenska** [comment]
medical main here, i don't like this change. the vast majority of patients who come through are people who have died either to natural hazards, and a little less commonly, getting killed by someone who just felt like killing. this punishes people who are playing the game normally much more than it does validhunters.

> This change should help antags facing unending tides of constantly-revived validhunters and make such players reconsider their life choices that led them to hunting changelings as an assistant with a blowtorch.

this statement disregards the many people who die completely unrelated to validhunting, while exaggerating the 'unending tides' that antags face. while, yes, the change does punish validhunters, but they're the least likely to wind up in medical to begin with. this moreso punishes the engineer who's trying to learn the sm for the first time, the shaft miner who got caught in an ash storm, the janitor who was gunned down in the middle of the hallway by a traitor who just wanted kills, etc.

> its not the antagonist's fault you performed what is commonly known a "skill issue" and died. You should value your life, its not disposable and this is a punishment for treating it as such.

there's an extremely high disparity in combat readiness between a traitor and a janitor. the janitor could be super robust, but the traitor has the equipment advantage. the traitor can just gun down the janitor. is that the janitor's fault that he didn't prepare for getting shot at while trying to clean? no. treating it like dying is a skill issue would only encourage powergaming, which is bad.

> this doesn't affect people actually playing med though

believe it or not, it does. no map has a dedicated recovery ward, and people will be afraid to go back out into the station. yes, we can shoo people out of medical (the amount of beds is very limited, after all), and they'll be more likely to come back into medical, dead, in the next ten minutes, compared to the otherwise identical spaceman who hasn't been revived.

and again, this mostly affects players who aren't playing to win. the atmos tech who got caught in an explosion while trying to fix a scientist's mistake in ordnance is now less capable of doing his job for ten minutes. the scientist who wanted to learn how to make bz explosives is waiting on the atmos tech to recover, effectively barred from doing his job. the engineer who wants experimental tools is now waiting on the atmos tech, in addition to the scientist, to finish the research for the tools. the assistant, who is friends with the engineer, is now also waiting on the atmos tech, the scientist, and the engineer, so the two of them can go out into space to build a hermit house for fun. meanwhile, the validhunter continues hunting down spies, with a pharmacy in his backpack so he'll never have to go to medical for revival.

yes, this is an exaggeration, but so is the issue highlighted in this pr.

**Comment by breadcrombs** [comment]
being dead is already punishing enough

**Comment by Admiral-Gigaton** [comment]
side note: valid hunters typically ARENT revived because they get fucking round removed or medical is swamped from bystanders getting shot and the antag coming in and blowing the doctors up, and after being blown up, they too now have the revival shock, making them more and more pissed, and eventually turning THEM into the RR valid hunters, increasing the load of dead people not getting revived because all the doctors got fucking nuked, twice, once by antag and once by revival shock fucking them up.

Testmerge only or not, NOTHING good can come out of this, especially if its testing whether or not we should turn this into a full PR to get fully merged

**Comment by Majorkite** [comment]
> > > > about in the middle of the set among all quirks
> > > 
> > > 
> > > That's pretty reasonably common. enough to suggest that if this was already a problem we would have seen the impact of the trait already.
> > 
> > 
> > This is enough data to make me think it might not be so bad. I still think the time and severity should be tweaked though. Like others have said 10 minutes is a whole shuttle call. Not an insignificant amount of time.
> 
> That is true but out of the 10 minutes you'll spend like 5 in medbay getting healed after revival, probably more if smart docs prioritize rezzing people to start the 10 min timer. Unless a doc stabs you during surgery or the fighting is in medbay you'll be fine with the extra wound chance as long as you play it semi safe. 10 min of -15 is a lot tho but i'm never sure about mood from just the numbers.

Five minutes getting healed after revival? If docs are smart, as you say, they're getting you to threshold and then zapping you and doing the rest of your damage with chems or sutures/mesh, maybe doing tend if they're lazy. That takes 2 minutes max, not five.  Only case where that would be mostly true is if the person has fully septic third/catastrophic burns cause then they gotta go in the tube. Which is already being put in additional timeout for dying anyway. 

My worries about the actual effects of the change are diminished slightly, but my big concern is still how people are going to _act_ when this pr gets pushed.

**Comment by MrMelbert** [comment]
We're maybe gonna set this up as a discord discussion if headcoders finish setting it up before 2025. If not then I guess we'll just testmerge it

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -887,6 +887,10 @@
 	// The signal is called after everything else so components can properly check the updated values
 	SEND_SIGNAL(src, COMSIG_LIVING_REVIVE, full_heal_flags)
 
+// If we have mood, applies a long-lasting mood debuff, majorly decreases our sanity and making us easier to injure. Does not affect changelings (duh)
+/mob/living/proc/revival_shock()
+	apply_status_effect(/datum/status_effect/revival_shock)
```
Comment (MrMelbert) [inline code/modules/mob/living/living.dm:892]: Probably doesn't need to be a proc anymo

File: code/game/objects/items/defib.dm [hunk 4]
```diff
@@ -510,171 +517,202 @@
...
+			span_userdanger("[user] touches [target] with [src]!"))
+	target.adjustStaminaLoss(60)
+	target.Knockdown(75)
+	target.set_jitter_if_lower(100 SECONDS)
+	target.apply_status_effect(/datum/status_effect/convulsing)
 	playsound(src,  'sound/machines/defib/defib_zap.ogg', 50, TRUE, -1)
-	if(HAS_TRAIT(M,MOB_ORGANIC))
-		M.emote("gasp")
-	log_combat(user, M, "zapped", src)
+	if(HAS_TRAIT(target, MOB_ORGANIC))
```
Comment (MrMelbert) [inline code/game/objects/items/defib.dm:540]: This isn't even a trait, wtf

File: code/game/objects/items/defib.dm [hunk 5]
```diff
@@ -510,171 +517,202 @@
...
+
+	if(!do_after(user, 1.5 SECONDS, target, extra_checks = CALLBACK(src, PROC_REF(is_wielded)))) //Takes longer due to overcharging
+		do_cancel()
+		return
+
+	if(!target)
+		do_cancel()
+		return
+
+	if(target && target.stat == DEAD)
```
Comment (MrMelbert) [inline code/game/objects/items/defib.dm:573]:
```suggestion
	if(target.stat == DEAD)
```

File: code/game/objects/items/defib.dm (hunk 4 above)
Comment (SmArtKar) [reply code/game/objects/items/defib.dm:540]: lmao i just fixed up the names but didn't notice this bit, yeah i'll fix this

**Comment by ExcessiveUseOfCobblestone** [comment]
I could probably be on board with this if there was a pretty heavy decay on the mood/sanity.

Generally speaking, I think I'm kinda in the camp where death itself is a punishment when u gotta wait to get healed, but there should be more permanent consequences that exponentially grow the more times you die so it punishes repeat offenders (combat seekers) while minimally hurting the guy who was just at the wrong place. Could probably use wounds to create a similar effect.

**Comment by SmArtKar** [comment]
People made some good points, I should get back to improving this

**Comment by Ghommie** [comment]
Is this going anywhere?

**Comment by SmArtKar** [comment]
> Is this going anywhere? I have been oblivious to the conversation on Discord but it would be nice to know something here too.

how has this not staled out wtf
